app/src/main/java/com/tohami/newsapi/ui/news/NewsFragment.java

```diff
@@ -0,0 +1,105 @@
+package com.tohami.newsapi.ui.news;
+
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+
+import androidx.annotation.Nullable;
+import androidx.databinding.DataBindingUtil;
+import androidx.lifecycle.ViewModelProviders;
+import androidx.navigation.Navigation;
+import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;
+
+import com.tohami.newsapi.R;
+import com.tohami.newsapi.adapter.NewsAdapter;
+import com.tohami.newsapi.base.BaseFragment;
+import com.tohami.newsapi.data.model.NewsArticle;
+import com.tohami.newsapi.databinding.FragmentNewsBinding;
+import com.tohami.newsapi.di.Injection;
+
+import java.util.ArrayList;
+
+public class NewsFragment extends BaseFragment implements NewsAdapter.OnItemClickedListener {
+
+    private NewsViewModel mViewModel;
+    private FragmentNewsBinding binding;
+
+
+    public static NewsFragment newInstance() {
+        return new NewsFragment();
+    }
+
+
+    @Override
+    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mViewModel = ViewModelProviders.of(getActivity(), Injection.provideViewModelFactory(getContext()))
+                .get(NewsViewModel.class);
+
+        NewsAdapter adapter = new NewsAdapter(new ArrayList<NewsArticle>() , this);
+        binding.rvNews.setAdapter(adapter);
+
+        mViewModel.getNewsArticlesLiveData().observe(this, listDataModel -> {
+            switch (listDataModel.getStatus()) {
+                case SUCCESS:
+                    binding.refreshLayout.setEnabled(true);
+                    binding.rvNews.setVisibility(View.VISIBLE);
+                    binding.errMsg.setVisibility(View.GONE);
+                    binding.bookloading.setVisibility(View.GONE);
+                    adapter.replaceItems(listDataModel.getResponseBody());
+                    break;
+                case FAIL:
+                    binding.refreshLayout.setEnabled(true);
+                    binding.rvNews.setVisibility(View.GONE);
+                    binding.errMsg.setVisibility(View.VISIBLE);
+                    binding.bookloading.setVisibility(View.GONE);
+                    binding.errMsg.setText(listDataModel.getErrorMsg());
+                    break;
+                case LOADING:
+                    binding.refreshLayout.setEnabled(false);
+                    binding.rvNews.setVisibility(View.GONE);
+                    binding.errMsg.setVisibility(View.GONE);
+                    binding.bookloading.setVisibility(View.VISIBLE);
+                    binding.bookloading.start();
+                    adapter.replaceItems(new ArrayList<>());
+                    break;
+                default:
+                    adapter.replaceItems(new ArrayList<>());
+            }
+        });
+
+        mViewModel.refreshNewsList();
```

Comment:
Flow should be as follows, view should subscribe to an observable returned from the view model. communication between the view and the view model should be through observables

Reply:
done